### DISCUSSION TITLE
support pnpm

### DISCUSSION THREAD

**Discussion by mrooding** [cawa-93/vite-electron-builder#1052] [Q&A]
**Is your feature request related to a problem? Please describe.**
After !1013 the repository wouldn't work anymore with pnpm

**Describe the solution you'd like**
Support using pnpm out of the box

**Describe alternatives you've considered**
N/A

**Additional context**
I was able to get pnpm to work by modifying a few things. However, unfortunately these changes break support for `npm`. I'm not sure whether there's a compatible method that supports both `npm` and `pnpm`. If there's not, maybe it's good enough to include support for package managers in the read me.

> npm error code EUNSUPPORTEDPROTOCOL
> npm error Unsupported URL Type "workspace:": workspace:../preload/

***pnpm workspace support***
First of all, the `workspaces` block in the root package.json isn't supported:

>  WARN  The "workspaces" field in package.json is not supported by pnpm. Create a "pnpm-workspace.yaml" file instead.

I created a `pnpm-workspaces.yaml` with the following content:

```yaml
packages:
  - "packages/*"
```

***workspaces dependencies***
I had to modify the way the workspaces are included as dev or regular dependencies. Instead of the `"*"` value, I had to use the `"workspace:../path" notation.

E.g.: `"@vite-electron-builder/main": "*"` became `"@vite-electron-builder/main": "workspace:./packages/main/"`

**Comment by cawa-93** [comment]
I don't really understand what exactly you are suggesting. The things you described are natural incompatibilities between the two tools. 
NPM expects workspace descriptions in a `package.json`, while [PNPN expects them in a separate file](https://github.com/pnpm/pnpm/issues/2255).
PNPM does support a special protocol, [NPN does not](https://github.com/npm/rfcs/issues/765). 

Supporting all versions of all tools is not a goal for this template. And if you want to switch one tool to another, you have to make small twiks.
